hudi-common/src/main/java/org/apache/hudi/common/model/debezium/AbstractDebeziumAvroPayload.java

```diff
@@ -91,4 +90,14 @@ private Option<IndexedRecord> handleDeleteOperation(IndexedRecord insertRecord)
   private Option<IndexedRecord> getInsertRecord(Schema schema) throws IOException {
     return super.getInsertValue(schema);
   }
+
+  @Override
+  protected boolean isDeleteRecord(GenericRecord record) {
+    return isDebeziumDeleteRecord(record) || super.isDeleteRecord(record);
+  }
+
+  private boolean isDebeziumDeleteRecord(GenericRecord record) {
```

Comment:
should we introduce something like 
```
isCustomDeleteRecord() 
```
in BaseAbroPayload 
which will by default return false. 

And impl of IsDeleteRecord in BaseAvroPayload will be 
```
protected boolean isDeleteRecord(GenericRecord record) {
    return isCustomDeleteRecord(record) || isDeleteRecordInternal(record);
  }
```

I am just trying to avoid calling super.isDeleteRecord in every custom payload. 

with this, every custom payload just needs to override isCustomDeleteRecord.

Reply:
Some payload may not want to leverage _hoodie_is_deleted.  Your code example will not allow this.

Reply:
sg